src/v/kafka/requests/kafka_batch_adapter.cc

```diff
@@ -123,22 +119,41 @@ void kafka_batch_adapter::verify_crc(int32_t expected_crc, iobuf_parser in) {
     }
 }
 
-void kafka_batch_adapter::adapt(iobuf&& kbatch) {
+iobuf kafka_batch_adapter::adapt(iobuf&& kbatch) {
```

Comment:
any reason to prefer this over returning a vector of batches or wrapping this parser in a record batch reader interface?

Reply:
I wanted to work a batch at a time for when streaming becomes a thing. I'll look at record batch reader interface.

Reply:
tbh we can leave this it looks good to me. we may need a streaming interface like the batch reader at some point if things get big to avoid stalls

Reply:
I'll add a streaming interface later on, but all of the serialisation is currently sync.